src/py/flwr/cli/login/login.py

```diff
@@ -0,0 +1,173 @@
+            "`options.num-supernodes` value).",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    # Validate the federation exists in the configuration
+    federation_config = config["tool"]["flwr"]["federations"].get(federation)
+    if federation_config is None:
+        available_feds = {
+            fed for fed in config["tool"]["flwr"]["federations"] if fed != "default"
+        }
+        typer.secho(
+            f"❌ There is no `{federation}` federation declared in "
+            "`pyproject.toml`.\n The following federations were found:\n\n"
+            + "\n".join(available_feds),
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    if "address" not in federation_config:
+        typer.secho(
+            f"❌ The federation `{federation}` does not have `SuperExec` "
+            "address in its config.\n Please specify the address in "
+            "`pyproject.toml` and try again.",
+            fg=typer.colors.RED,
+            bold=True,
+        )
+        raise typer.Exit(code=1)
+
+    stub = _create_exec_stub(app, federation_config)
+    login_request = LoginRequest()
+    login_response: LoginResponse = stub.Login(login_request)
+    auth_plugin = auth_plugins[login_response.login_details.get("auth_type", "")]
+    config = auth_plugin.login(
+        dict(login_response.login_details), config, federation, stub
+    )
+
+    base_path = get_flwr_dir()
+    credentials_dir = base_path / ".credentials"
+    credentials_dir.mkdir(parents=True, exist_ok=True)
+
+    credential = credentials_dir / federation_config["address"]
+
+    with open(credential, "wb") as config_file:
+        dump(config, config_file)
+
+
+def _create_exec_stub(app: Path, federation_config: dict[str, Any]) -> ExecStub:
+    insecure_str = federation_config.get("insecure")
+    if root_certificates := federation_config.get("root-certificates"):
+        root_certificates_bytes = (app / root_certificates).read_bytes()
+        if insecure := bool(insecure_str):
+            typer.secho(
+                "❌ `root_certificates` were provided but the `insecure` parameter"
+                "is set to `True`.",
+                fg=typer.colors.RED,
+                bold=True,
+            )
+            raise typer.Exit(code=1)
+    else:
+        root_certificates_bytes = None
+        if insecure_str is None:
+            typer.secho(
+                "❌ To disable TLS, set `insecure = true` in `pyproject.toml`.",
+                fg=typer.colors.RED,
+                bold=True,
+            )
+            raise typer.Exit(code=1)
+        if not (insecure := bool(insecure_str)):
+            typer.secho(
+                "❌ No certificate were given yet `insecure` is set to `False`.",
+                fg=typer.colors.RED,
+                bold=True,
+            )
+            raise typer.Exit(code=1)
```

Comment:
Similar as above, could you use `validate_certificate_in_federation_config` function instead?